packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -76,12 +106,27 @@ pub enum TransactionType {
     Create(CreateTransaction),
 }
 
-pub trait Transaction: Into<FuelTransaction> + Clone {
+pub trait EstimablePredicates {
+    /// If a transactions contains predicates, we have to estimate them
+    /// before sending the transaction to the node. The estimation will check
+    /// all predicates and set the `predicate_gas_used` to the actual consumed gas.
```

Comment:
```suggestion
    /// If a transaction contains predicates, we have to estimate them
    /// before sending the transaction to the node. The estimation will check
    /// all predicates and set the `predicate_gas_used` to the actual consumed gas. 
```

Excellent comment btw!